lib/src/strategies.dart

```diff
@@ -54,7 +55,16 @@ class _ShareableConnectionsStrategy implements _Strategy {
       connDestroyer(conn.conn);
     }
   }
-  
+
+  Future closeConnections(_ConnDestroyer connDestroyer) {
+    return Future.forEach(_pool, (futureConn) {
```

Comment:
use a `for .. in` with await. And await for each connection close. Otherwise awaiting on this doesn't give any benefit.